txpipe/twopoint_fourier.py

```diff
@@ -348,137 +358,177 @@ def select_calculations(self, nbins_source, nbins_lens):
             for j in range(nbins_lens):
                 calcs.append((i, j, k))
 
-        # For position-position we omit pairs with j<i
+        # For position-position we omit pairs with j>i.
+        # We do keep cross-pairs, since even though we may not want to
+        # do parameter estimation with them they are useful diagnostics.
         k = POS_POS
         for i in range(nbins_lens):
             for j in range(i + 1):
                 calcs.append((i, j, k))
 
         return calcs
 
-    def compute_power_spectra(self, pixel_scheme, i, j, k, f_wl, f_d, w00, w02, w22, ell_bins, noise, cl_theory):
+    def compute_power_spectra(self, pixel_scheme, i, j, k, f_wl, f_d, ell_bins, tomo_info, cl_theory):
         # Compute power spectra
         # TODO: now all possible auto- and cross-correlation are computed.
         #      This should be tunable.
         # TODO: Fix window functions, and how to save them.
 
         # k refers to the type of measurement we are making
         import sacc
-        import pymaster
+        import pymaster as nmt
+        import healpy
+
         CEE=sacc.standard_types.galaxy_shear_cl_ee
         CBB=sacc.standard_types.galaxy_shear_cl_bb
         CdE=sacc.standard_types.galaxy_shearDensity_cl_e
         CdB=sacc.standard_types.galaxy_shearDensity_cl_b
         Cdd=sacc.standard_types.galaxy_density_cl
 
         type_name = NAMES[k]
-        print(f"Process {self.rank} calcluating {type_name} spectrum for bin pair {i},{i}")
+        print(f"Process {self.rank} calculating {type_name} spectrum for bin pair {i},{j}")
         sys.stdout.flush()
 
-        # We need the theory spectrum for this pair
-        theory = cl_theory[(i,j,k)]
 
         # The binning information - effective (mid) ell values and
         # the window information
         ls = ell_bins.get_effective_ells()
         win = [ell_bins.get_window(b) for b,l  in enumerate(ls)]
 
+        # The healpix pixel windows.  C_ell estimates on healpix
+        # maps (like all pixelized maps) are modulated by a pixel
+        # window function, which we calculate here so we can remove
+        # it below.  These are trivially fast to get, so no point
+        # caching.
+        if (pixel_scheme.name == 'healpix') and self.config['pixel_window']:
+            # Get the raw pixel window functions
+            pixwin_t, pixwin_p = healpy.pixwin(pixel_scheme.nside, True)
+            # Interpolate to our ell values
+            pixwin_ell = np.arange(pixwin_t.size)
+            pixwin_t = np.interp(ls, pixwin_ell, pixwin_t)
+            pixwin_p = np.interp(ls, pixwin_ell, pixwin_p)
+        else:
+            # TODO figure out what these are supposed to be
+            pixwin_t = 1.0
+            pixwin_p = 1.0
+
+        # We need the theory spectrum for this pair
+        #TODO: when we have templates to deproject, use this.
```

Comment:
Just checking: you've not deprojected anything in the plots you sent me on slack, right?

Reply:
Nope.